lib/index.js

```
@@ -12,13 +12,18 @@ const Wreck = require('wreck');
 const internals = {
     defaults: {
         threshold: 20,
+        errorThreshold: 0,
         schema: 'good-http',
         wreck: {
             timeout: 60000,
             headers: {}
         }
     },
-    host: Os.hostname()
+    host: Os.hostname(),
+    shouldHoldData(errorThreshold, failureCount) {
+
+        return errorThreshold !== null && failureCount < errorThreshold;
```

If we set it to `Infinity` internally then we'd still need to do an explicit check for `Infinity` here and on line 61 in order to continue to support this behavior: "Set to `null` to ignore all errors and always clear events".